## Serialization || C++ || Python3

# unsigned 8-bit int || uint8_t || bool
bool boolean_value_bool

# signed 8-bit / 16-bit / 32-bit int || int8_t / int16_t / int32_t || int
int8 int8_value_int
int16 int16_value_int
int32 int32_value_int

# signed 64-bit int || int64_t || long
int64 int64_value_long_int

# unsigned 8-bit / 16-bit / 32-bit int || uint8_t / uint16_t / uint32_t || int
uint8 uint8_value_unsigned_int
uint8 uint16_value_unsigned_int
uint8 uint32_value_unsigned_int

# unsigned 64-bit int || uint64_t || long
uint8 uint64_value_unsigned_long_int

# 32-bit IEEE float || float || float
float32 float32_value_float

# 64-bit IEEE float || double || float
float64 float64_value_double

# ascii string || std::string || str
string string_value_std_string


##################################################
##                   Arrays                     ##
##################################################

# variable-lenght || std::vector<Type> || list of Types
int32[] array_value_std_vector_int
float64[] array_value_std_vector_double
string[] array_value_std_vector_string


##################################################
##                  Costants                    ##
##################################################

# constanttype CONSTANTNAME=constantvalue
int32 CONSTANT1=0
string CONSTANT2="EXAMPLE"
